File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        
        int maxwater = 0;
        int n = height.size();
        int water = 0;
        int left = 0;
        int right = n - 1;
        while(left < right){
            water = min(height[left],height[right]) * (right - left);
            maxwater = max(maxwater, water);
            if(height[left] < height[right]){
                left++;
            }else{
                right--;
            }
        }
        return maxwater;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int n=height.length;
        int maxarea=Integer.MIN_VALUE;
        int area=1;
        int i=0;
        int j=n-1;
        while(i<j){
            if(height[i]<=height[j]){
                int width=j-i;
                area=height[i]*width;
                i++;
            }else{
                int width=j-i;
                area=height[j]*width;
                j--;
            }
            maxarea=Math.max(area,maxarea);
        }

        return maxarea;

    }
}